src/client/public/scripts/pages/github-app-config.ts

```diff
@@ -0,0 +1,202 @@
+
+    // tslint:disable-next-line:no-any
+    const responseBody =
+        await response.json() as api.JSONAPIResponse<api.OrgSettings>;
+    if ('error' in responseBody) {
+      this.errorMsg = responseBody.error.message;
+      return null;
+    }
+
+    if (!('data' in responseBody)) {
+      this.errorMsg = 'No data provided for installation ID.';
+      return null;
+    }
+
+    return responseBody.data.fileContents;
+  }
+
+  async getOrgNameFromInstallId(installId: string): Promise<string|null> {
+    const response = await fetch('/api/github-app/details', {
+      method: 'POST',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify({
+        installId: Number(installId),
+      }),
+      credentials: 'include',
+    });
+
+    const responseBody =
+        await response.json() as api.JSONAPIResponse<api.GithubAppInstall>;
+    if ('error' in responseBody) {
+      this.errorMsg = responseBody.error.message;
+      return null;
+    }
+
+    if (!('data' in responseBody)) {
+      this.errorMsg = 'No data provided for installation ID.';
+      return null;
+    }
+
+    const installData = responseBody.data;
+    return installData.login;
+  }
+
+  render(): TemplateResult {
+    if (this.errorMsg) {
+      return html`<h2>GitHub App Configuration</h2>
+
+      <p>${this.errorMsg}</p>`;
+    }
+
+    if (!this.initialised) {
+      return html`<h2>GitHub App Configuration</h2>
+
+      <p>Loading....</p>`;
+    }
+
+    let configDisplay = DEFAULT_CONFIG;
+    if (this.currentConfig) {
+      configDisplay = this.currentConfig;
+    }
+
+    return html`
+<h2>GitHub App Configuration</h2>
+<div class="editor-container">
+<textarea class="github-app-defaults text-editor">
+{
+  // Priority issues. Enable this to add support
+  // for priority labels.
+  "issues.priorityIssues": false,
+}
+</textarea>
+<textarea id="settings-text-area" class="text-editor">
+${configDisplay}
+</textarea>
+</div>
+<div class="button-container">
+  <button id="reset-btn" on-click="${() => this.resetConfig()}">Reset</button>
+  <button id="save-btn" on-click="${() => this.saveConfig()}">Save</button>
+</div>
+`;
+  }
+
+  private async saveConfig() {
+    const settingsTextArea = document.querySelector('#settings-text-area');
+    const resetButton = document.querySelector('#reset-btn');
+    const saveButton = document.querySelector('#save-btn');
+    if (!settingsTextArea) {
+      console.warn('Unable to find the \'#settings-text-area\' element.');
+      return;
+    }
+    if (!resetButton) {
+      console.warn('Unable to find the \'#reset-btn\' element.');
+      return;
+    }
+    if (!saveButton) {
+      console.warn('Unable to find the \'#save-btn\' element.');
+      return;
+    }
+
+    settingsTextArea.setAttribute('disabled', 'true');
+    resetButton.setAttribute('disabled', 'true');
+    saveButton.setAttribute('disabled', 'true');
+
+    const response = await fetch('/api/github-app/save-config', {
```

Comment:
Yeah, I would probably decouple this from `github-app`. As above, I would prefer this to be `POST /api/org/:org`. Orgs have configs which we'll always keep around. The GitHub app will read this config, but they could conceivably be separate. eg. Maybe I re-install the app - it should keep the config though.